packages/mux-video/src/CustomVideoElement.js

```diff
@@ -69,7 +69,26 @@ class CustomVideoElement extends HTMLElement {
     this.attachShadow({ mode: 'open' });
     this.shadowRoot.appendChild(template.content.cloneNode(true));
 
-    const nativeEl = (this.nativeEl = this.shadowRoot.querySelector('video'));
+    const media = this.shadowRoot.querySelector('[name=media]');
+    const nativeEl = (this.nativeEl = media.assignedElements({ flatten: true })[0]);
+
+    if (nativeEl.hasAttribute('is')) {
+      customElements.whenDefined(nativeEl.getAttribute('is')).then(() => {
+        // If nativeEl is a custom built-in forward new methods and props.
+        const props = Object.getOwnPropertyNames(Object.getPrototypeOf(nativeEl));
```

Comment:
non-blocking/followup: this is probably going to be a bit brittle, since it assumes the relevant props are on:

1. the prototype of the element and not the instance (safer for native HTMLMediaElement and related)
2. the immediate prototype of the element and not somewhere in the protochain (a bigger concern if we're anticipating e.g. mixins or other inheritance chains)